File: backend/node/routes/studybotRoutes.js
const express = require('express');
const router = express.Router();

router.post('/chat', (req, res) => {
  const { username, message } = req.body;
  
  const reply = "I can help you create a study plan! Use the Quick Plan Creator on the right to get started.";
  
  res.json({ reply });
});

router.post('/plan', (req, res) => {
  const { username, examName, startDate, examDate, dailyHours, topics } = req.body;
  
  const start = new Date(startDate);
  const exam = new Date(examDate);
  const totalDays = Math.ceil((exam - start) / (1000 * 60 * 60 * 24));
  
  const phases = [
    { name: "Foundation", startDay: 1, endDay: Math.floor(totalDays * 0.4), focus: "Learn basics and fundamentals" },
    { name: "Practice", startDay: Math.floor(totalDays * 0.4) + 1, endDay: Math.floor(totalDays * 0.8), focus: "Solve problems and practice" },
    { name: "Revision", startDay: Math.floor(totalDays * 0.8) + 1, endDay: totalDays, focus: "Review and mock tests" }
  ];
  
  const dailyPlan = [];
  for (let i = 1; i <= Math.min(totalDays, 10); i++) {
    const currentDate = new Date(start);
    currentDate.setDate(start.getDate() + i - 1);
    
    dailyPlan.push({
      dayIndex: i,
      date: currentDate.toISOString().split('T')[0],
      tasks: topics.slice(0, 2).map(t => `Study ${t.name} (${dailyHours/2} hours)`)
    });
  }
  
  const plan = {
    examName,
    startDate,
    examDate,
    totalDays,
    dailyHours,
    topics,
    phases,
    dailyPlan
  };
  
  res.json({
    message: `Study plan created for ${examName}! ${totalDays} days, ${dailyHours} hours/day.`,
    plan
  });
});

module.exports = router;

File: backend/python/routes/studybot_routes.py
from flask import Blueprint, request, jsonify
from datetime import datetime, timedelta

studybot_bp = Blueprint('studybot', __name__)

@studybot_bp.route('/chat', methods=['POST'])
def chat():
    data = request.json
    reply = "I can help you create a study plan! Use the Quick Plan Creator on the right to get started."
    return jsonify({'reply': reply})

@studybot_bp.route('/plan', methods=['POST'])
def create_plan():
    data = request.json
    username = data.get('username')
    exam_name = data.get('examName')
    start_date = data.get('startDate')
    exam_date = data.get('examDate')
    daily_hours = data.get('dailyHours')
    topics = data.get('topics', [])
    
    start = datetime.strptime(start_date, '%Y-%m-%d')
    exam = datetime.strptime(exam_date, '%Y-%m-%d')
    total_days = (exam - start).days
    
    phases = [
        {'name': 'Foundation', 'startDay': 1, 'endDay': int(total_days * 0.4), 'focus': 'Learn basics and fundamentals'},
        {'name': 'Practice', 'startDay': int(total_days * 0.4) + 1, 'endDay': int(total_days * 0.8), 'focus': 'Solve problems and practice'},
        {'name': 'Revision', 'startDay': int(total_days * 0.8) + 1, 'endDay': total_days, 'focus': 'Review and mock tests'}
    ]
    
    daily_plan = []
    for i in range(1, min(total_days + 1, 11)):
        current_date = start + timedelta(days=i - 1)
        tasks = [f"Study {t['name']} ({daily_hours/2} hours)" for t in topics[:2]]
        daily_plan.append({
            'dayIndex': i,
            'date': current_date.strftime('%Y-%m-%d'),
            'tasks': tasks
        })
    
    plan = {
        'examName': exam_name,
        'startDate': start_date,
        'examDate': exam_date,
        'totalDays': total_days,
        'dailyHours': daily_hours,
        'topics': topics,
        'phases': phases,
        'dailyPlan': daily_plan
    }
    
    return jsonify({
        'message': f'Study plan created for {exam_name}! {total_days} days, {daily_hours} hours/day.',
        'plan': plan
    })
